### PR TITLE
feat: initial setup for icons library

### PR DESCRIPTION
**Pull request by rahulkeswani101**
part of https://github.com/SigNoz/engineering-pod/issues/1352

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,9 @@
+MIT License
+
+Copyright (c) 2024 Rahul Keswani
```
Comment (YounixM) [inline LICENSE]: `SigNoz`

File: dist/index.esm.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
+import _extends from '@babel/runtime/helpers/extends';
```
Comment (YounixM) [inline dist/index.esm.js]: This should be generated during the build process

File: assets/heart-filled.svg
Comment (ahmadshaheer) [inline assets/heart-filled.svg:1]: nit: have we compressed the SVGs?

File: dist/index.esm.js
Comment (ahmadshaheer) [inline dist/index.esm.js:1]: can you add the dist folder to `.gitignore` file?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    "React",
+    "SVG",
+    "Icons"
+  ],
+  "module": "dist/index.esm.js",
+  "files": [
+    "dist"
+  ],
+  "scripts": {
+    "prebuild": "rimraf src && rimraf dist",
```
Comment (ahmadshaheer) [inline package.json:15]: will `rm -rf src && rm -rf dist` not work here?

File: package.json
Comment (ahmadshaheer) [inline package.json:1]: How are we going to manage the version bumps for this package? can you use [changesets](https://github.com/changesets/changesets)? it can make the version and release management easier.

It can be taken in a separate PR as well, IMO.

File: assets/heart-filled.svg
Comment (rahulkeswani101) [reply assets/heart-filled.svg:1]: We are using SVGO for this purpose.

File: package.json
Comment (rahulkeswani101) [reply package.json:1]: I am considering taking it in the next PR but let's take input from other team members.

File: package.json (hunk 3 above)
Comment (rahulkeswani101) [reply package.json:15]: @ahmadshaheer This will work but rimraf is often preferred because it's a cross-platform solution.

**Review by ahmadshaheer — DISMISSED** [review]
Thanks for making the changes, @rahulkeswani101. 

a minor issue, I noticed that `dist/index.esm.js` is still present in the Files changed.

**Review by ahmadshaheer — DISMISSED** [review]

File: package.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "build": "npm run svgr && rollup -c",
+    "svgr": "svgr --icon --title-prop --replace-attr-values '#1D1D1D=currentColor' -d src -- assets"
+  },
+  "author": "Rahul Keswani",
```
Comment (vikrantgupta25) [inline package.json]: SigNoz

File: dist/index.esm.js (hunk 2 above)
Comment (rahulkeswani101) [reply dist/index.esm.js]: This has been removed already.

**Review by YounixM — DISMISSED** [review]

File: package.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "@signoz-icons/icons",
+  "version": "1.0.0",
```
Comment (vikrantgupta25) [inline package.json]: shouldn't we start with `0.0.0` version and then keep the version number increments on top of it? `1.0.0` is generally used when the project is marked as stable and has achieved a decent good state.

Comment (YounixM) [reply package.json]: @YounixM : yes .. We can start with minor / patch version update till we add all icons and then do a `1.0.0`

Comment (YounixM) [reply package.json]: @rahulkeswani101 : Please take care of this.

Comment (vikrantgupta25) [reply package.json]: https://semver.org/#spec-item-4

**Review by vikrantgupta25 — APPROVED** [review]

**Review by YounixM — APPROVED** [review]
